client/src/extension.ts

```diff
@@ -235,21 +233,14 @@ export function activate(context: ExtensionContext) {
   // Start the client. This will also launch the server
   client.start();
 
-  // Restart the language client automatically when certain configuration
-  // changes. These are typically settings that affect the capabilities of the
-  // language client, and because of that requires a full restart.
   context.subscriptions.push(
     workspace.onDidChangeConfiguration(({ affectsConfiguration }) => {
-      // Put any configuration that, when changed, requires a full restart of
-      // the server here. That will typically be any configuration that affects
-      // the capabilities declared by the server, since those cannot be updated
-      // on the fly, and require a full restart with new capabilities set when
-      // initializing.
       if (
-        affectsConfiguration("rescript.settings.inlayHints") ||
-        affectsConfiguration("rescript.settings.codeLens")
+        affectsConfiguration("rescript.settings")
       ) {
-        commands.executeCommand("rescript-vscode.restart_language_server");

```

Comment:
Did you test this in an actual editor? Since this affects what capabilities the LS itself is initiated with, I can't imagine turning features _on_ still works after this change, since you'd need to update the capabilities, and to do that you must restart the server.